File: project_euler/00/002.go
// Copyright [2017] <dmnsn7@gmail.com>

package main

import "fmt"

const d = 4000000

func main() {
	fib := []int{1, 2}
	for fib[len(fib)-2]+fib[len(fib)-1] <= d {
		fib = append(fib, fib[len(fib)-2]+fib[len(fib)-1])
	}

	sum := 0
	for i := 0; i < len(fib); i++ {
		if fib[i]%2 == 0 {
			sum += fib[i]
		}
	}

	fmt.Printf("%d\n", sum)
}

File: project_euler/00/002.py
#!/usr/bin/env python3
'''Copyright [2017] <dmnsn7@gmail.com>'''

D = 4000000


def main():
    fib = [1, 2]
    while fib[-2] + fib[-1] <= D:
        fib.append(fib[-2] + fib[-1])

    sum_up = 0
    for i, _ in enumerate(fib):
        if fib[i] % 2 == 0:
            sum_up += fib[i]

    print(sum_up)


if __name__ == '__main__':
    main()
